Update Readme (#227)
eProsima/Micro-XRCE-DDS-Agent: README.md

[![Forks](https://img.shields.io/github/forks/eProsima/Micro-XRCE-DDS-Agent.svg)](https://github.com/eProsima/Micro-XRCE-DDS-Agent/network/members)
[![Stars](https://img.shields.io/github/stars/eProsima/Micro-XRCE-DDS-Agent.svg)](https://github.com/eProsima/Micro-XRCE-DDS-Agent/stargazers)

[![Docker Build Status](https://img.shields.io/docker/cloud/build/eprosima/micro-xrce-dds-agent)](https://hub.docker.com/r/eprosima/micro-xrce-dds-agent/)

<a href="http://www.eprosima.com"><img src="https://encrypted-tbn3.gstatic.com/images?q=tbn:ANd9GcSd0PDlVz1U_7MgdTe0FRIWD0Jc9_YH-gGi0ZpLkr-qgCI6ZEoJZ5GBqQ" align="left" hspace="8" vspace="2" width="100" height="100" ></a>

> *Micro XRCE-DDS Agent* acts as a server between the DDS dataspace network and *Micro XRCE-DDS Client* applications.

Reviewer: ```suggestion
*The Micro XRCE-DDS Agent* acts as a server to bridge the DDS dataspace network with *Micro XRCE-DDS Client* applications.
```

> *Micro XRCE-DDS Agents* receive messages containing request operations from *Clients* to publish and subscribe to/from the DDS global dataspace.

Reviewer: ```suggestion
The *Micro XRCE-DDS Agents* receive messages containing request operations from the *Clients* to publish and subscribe to topics in the DDS global dataspace. Remote procedure calls, as defined by the DDS-RPC standard, are also supported, allowing to communicate according to a request/reply paradigm.
```

> The *Agents* then process these XRCE requests and send back a response with the operation status result and with the requested data, in the case of subscribe operations.

Reviewer: ```suggestion
The *Agents* then process these XRCE requests and send back a response with the operation status result and with the requested data, in the case of subscribe/request/reply operations.
```

> Remote procedure calls, as defined by the DDS-RPC standard, are also supported.

Reviewer: ```suggestion
```

> *Agents* keep track of the *Clients* and the *Micro XRCE-DDS Entities* they create, by means of a dedicated *ProxyClient* entity that acts on behalf of them.

Reviewer: ```suggestion
*Agents* keep track of the *Clients* by means of a dedicated *ProxyClient* entity that acts on behalf of the latter.
```

> To do so, the *Agent* uses the *XRCE Entities* to interact with the DDS Global dataspace.

Reviewer: ```suggestion
This is made possible by the creation of *DDS Entities* on the *Agent* as a result of *Clients*' operations, such as *Participants*, *Topics*, *Publishers*, and *Subscribers*, which can interact with the DDS Global dataspace.
```

> The communication between a *Micro XRCE-DDS Client* and a *Micro XRCE-DDS Agent* is achieved by means of several kinds of built-in transports: **UDPv4**, **UDPv6**, **TCPv4**, **TCPv6** and **Serial** communication.

Reviewer: ```suggestion
The communication between a *Micro XRCE-DDS Client* and a *Micro XRCE-DDS Agent* is achieved by means of several kinds of built-in transports: **UDPv4**, **UDPv6**, **TCPv4**, **TCPv6** and **Serial** communication. In addition, there is the possibility for the user to generate its own **Custom** transport.
```

> You can use an *Agent* with these transports by means of the standalone executable generated when building the project, which comes with a built-in CLI tool to select one among the list previously mentioned.

Reviewer: ```suggestion
You can use an *Agent* with these transports by means of the standalone executable generated when building the project, which comes with a built-in CLI tool to select one of the transports listed above.
```
